### PR TITLE
Make convertToType more generic, and reduce empty value console warnings

### PR DESCRIPTION
**Pull request by nielsm5**
Make `ClassUtils#convertToType` more generic
Reduce empty value console warnings
Add tests to cover #2763
Revert iaf-test jms queue suffix removal #4749

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java [hunk 1]
```diff
@@ -299,7 +301,7 @@ public static String classNameOf(Object o) {
 	 */
 	public static void invokeSetter(Object bean, Method method, String valueToSet) {
 		if(!method.getName().startsWith("set") || method.getParameterTypes().length != 1) {
-			throw new IllegalArgumentException("method must start with [set] and may only contain [1] parameter");
+			throw new IllegalStateException("method must start with [set] and may only contain [1] parameter");
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java:304]: I'd argue that `IllegalArgumentException` was more appropriate, what was the motivation to change?

File: core/src/test/java/nl/nn/adapterframework/util/ClassUtilsTest.java [hunk 2]
```diff
@@ -311,19 +311,23 @@ private IScopeProvider getBytesClassLoaderProvider() throws Exception {
...
+		assertEquals("7", ClassUtils.convertToType(String.class, "7"));
+		assertEquals(true, ClassUtils.convertToType(boolean.class, "true"));
+		assertEquals(true, ClassUtils.convertToType(Boolean.class, "true"));
+		assertEquals(false, ClassUtils.convertToType(Boolean.class, "niet true"));
+		assertEquals(TestEnum.ONE, ClassUtils.convertToType(TestEnum.class, "one"));
+
+		assertThrows(IllegalArgumentException.class, ()->ClassUtils.convertToType(Object.class, "dummy"));
+		assertThrows(IllegalArgumentException.class, ()->ClassUtils.convertToType(Long.class, "dummy"));
+
+		assertThrows(IllegalArgumentException.class, ()->ClassUtils.convertToType(int.class, "")); //Empty string
```
Comment (tnleeuw) [inline core/src/test/java/nl/nn/adapterframework/util/ClassUtilsTest.java:331]: Could be wrapped in a call to `assertAll`

**Review by tnleeuw — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java (hunk 1 above)
Comment (nielsm5) [reply core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java:304]: I agree, however in this case the IllegalArgumentExceptions are meant for the `valueToSet` and not the `method` being invalid.

File: core/src/test/java/nl/nn/adapterframework/util/ClassUtilsTest.java (hunk 2 above)
Comment (nielsm5) [reply core/src/test/java/nl/nn/adapterframework/util/ClassUtilsTest.java:331]: Ah ja, nog een beetje nieuw voor mij.

**Pull request merged**